kolibri/content/api.py

```diff
@@ -98,19 +96,31 @@ class Meta:
         fields = ['parent', 'search', 'prerequisite_for', 'has_prerequisite', 'related',
                   'recommendations_for', 'next_steps', 'popular', 'resume', 'ids', 'content_id', 'channel_id', 'kind']
 
-    def title_description_filter(self, queryset, name, value):
+    def filter_search(self, queryset, name, value):
         """
-        search for title or description that contains the keywords that are not necessary in adjacent
+        Implement various filtering strategies in order to get a wide range of search results.
         """
-        exact_match = queryset.filter(Q(parent__isnull=False), Q(title__icontains=value) | Q(description__icontains=value))
-        if exact_match:
-            return exact_match
-        # if no exact match, fuzzy search using the stemmed_metaphone field in ContentNode that covers the title and description
-        fuzzed_tokens = [fuzz(word) for word in value.split()]
-        if not fuzzed_tokens[0]:
-            return []
-        token_queries = [reduce(lambda x, y: x | y, [Q(stemmed_metaphone__contains=token) for token in tokens]) for tokens in fuzzed_tokens]
-        return queryset.filter(reduce(lambda x, y: x & y, token_queries))
+
+        # search the whole string in the title or description
+        query = Q(title__icontains=value) | Q(description__icontains=value)
+
+        # split string up by spaces and search each individual string in title or description
+        str_arr = value.split()
+        for string in str_arr:
+            query.add(Q(title__icontains=string) | Q(description__icontains=string), Q.OR)
+        matches = queryset.filter(query)
+
+        # remove duplicate content items based on content_id
+        content_id_set = set()
+        id_list = []
+        for id_pair in matches.values('content_id', 'id')[:30]:
```

Comment:
looks like this will truncate the number of results _before_ de-duplication rather than after, so you'll end up with fewer than 30 results